src/app.js

```diff
@@ -6,8 +6,294 @@ import _ from 'underscore';
+    })
+  }
+  $('#status-messages').show();
+};
+
+const updateStatusMessageWith = (message) => {
+  $('#status-messages ul').empty();
+  $('#status-messages ul').append(`<li>${message}</li>`);
+  $('#status-messages').show();
+};
+
+const renderTrips = function renderTrips(tripList) {
+//   iterate through the tripList, generate HTML
+//   for each model and attatch it to the DOM
+  const tripTableElement = $('#trip-list');
+  tripTableElement.html('');
+
+  tripList.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripTableElement.append(generatedHTML);
+  });
+  console.log('ran render');
+  console.dir(tripList);
+
+  // SORTING
+  // Provide visual feedback for sorting
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${tripList.comparator}`).addClass('current-sort-field');
+  // alert(`sorted by ${tripList.comparator}`);
+};
+
+
+let singleTripTemplate;
+
+const renderSingleTrip = function renderSingleTrip(trip) {
+  const tripElement = $('#trip-detail');
+  const generatedHTML = singleTripTemplate(trip.attributes);
+  tripElement.html(generatedHTML);
+
+    console.log("checking this");
+}
+
+let successfullSave = function(model, response) {
+  updateStatusMessageWith(`${model.get('name')} added!`)
+};
+let failedSave = function(model, response) {
+  updateStatusMessageFrom(response.responseJSON.errors);
+  model.destroy();
+};
+
+//ADDING A TRIP
+const addTripHandler = function(event) {
+  event.preventDefault();
+
+  const tripData = {};
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+    const inputElement = $(`#add-trip-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    tripData[field] = value;
+
+    inputElement.val('');
+  });
+
+  console.log("Read trip data");
+  console.log(tripData);
+
+  const trip = new Trip(tripData);
+
+  if (trip.isValid()) {
+      tripList.add(trip);
+      trip.save({}, {
+        success: successfullSave,
+        error: failedSave,
+      });
+  } else {
+      // getting here means there were client-side validation errors reported
+      updateStatusMessageFrom(trip.validationError);
+  }
+
+  setTimeout(function(){ $('#status-messages').hide(); }, 10000);
+
+};
+
+//ADDING A Reservation
+const addReservationHandler = function(event) {
+  event.preventDefault();
+  const reservationData = {};
+  ['name', 'email', 'age'].forEach((field) => {
+    // select the input corresponding to the field we want
+    const inputElement = $(`#reservation-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    reservationData[field] = value;
+
+    inputElement.val('');
+  });
+
+  console.log("Read Reservation data");
+  console.log(reservationData);
+
+  reservationData["trip_id"] = $(this).data("id");
+  const reservation = new Reservation(reservationData);
+
+  console.log(reservation.url);
+
+  // reservation.save({}, {
+  if (reservation.save) {
```

Comment:
This doesn't actually call the reservation's `save` function.  Instead it checks to see if it exists.  

Remember this isn't Rails!  

Instead you need:
```
reservation.save({} {
  success: someEventHandler,
  error: someOtherEventHandler
});
```